lib/rules/no-var.js

```diff
@@ -17,19 +83,63 @@ module.exports = {
             recommended: false
         },
 
-        schema: []
+        schema: [],
+        fixable: "code"
     },
 
     create: function(context) {
+        var sourceCode = context.getSourceCode();
+
+        /**
+         * Checks whether it can fix a given variable declaration or not.
+         * It cannot fix if the following cases:
+         *
+         * - A variable is declared on a SwitchCase node.
+         * - A variable is redeclared.
+         * - A variable is used from outside the scope.
+         *
+         * @param {ASTNode} node - A variable declaration node to check.
+         * @returns {boolean} `true` if it can fix the node.
+         */
+        function canFix(node) {
+            var variables = context.getDeclaredVariables(node);
+            var scopeNode = getScopeNode(node);
+
+            return !(
+                node.parent.type === "SwitchCase" ||
```

Comment:
Can you add a comment to this line explaining why we are skipping SwitchCase. Otherwise next person who fixes the rule might think it was mistake.